src/main/groovy/com/twcable/grabbit/jcr/AuthorizableProtoNodeDecorator.groovy

```diff
@@ -0,0 +1,282 @@
+package com.twcable.grabbit.jcr
+
+import com.twcable.grabbit.proto.NodeProtos.Node as ProtoNode
+import com.twcable.grabbit.security.AuthorizablePrincipal
+import com.twcable.grabbit.security.InsufficientGrabbitPrivilegeException
+import com.twcable.grabbit.util.CryptoUtil
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import java.lang.reflect.Field
+import java.lang.reflect.Method
+import java.lang.reflect.ReflectPermission
+import javax.annotation.Nonnull
+import javax.jcr.Session
+import org.apache.jackrabbit.api.security.user.Authorizable
+import org.apache.jackrabbit.api.security.user.Group
+import org.apache.jackrabbit.api.security.user.User
+import org.apache.jackrabbit.api.security.user.UserManager
+import org.apache.jackrabbit.value.StringValue
+import org.apache.sling.jcr.base.util.AccessControlUtil
+
+/*
+ * Copyright 2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This class wraps a serialized node that represents an Authorizable. Authorizables are special system protected nodes, that can only be written under certain
+ * trees, and can not be written directly by a client.
+ */
+@CompileStatic
+@Slf4j
+class AuthorizableProtoNodeDecorator extends ProtoNodeDecorator {
+
+
+    protected AuthorizableProtoNodeDecorator(@Nonnull ProtoNode node, @Nonnull Collection<ProtoPropertyDecorator> protoProperties) {
+        this.innerProtoNode = node
+        this.protoProperties = protoProperties
+    }
+
+
+    @Override
+    JcrNodeDecorator writeToJcr(@Nonnull Session session) {
+        if(!checkSecurityPermissions()) {
+            throw new InsufficientGrabbitPrivilegeException("JVM Permissions needed by Grabbit to sync Users/Groups were not found. See log for specific permissions needed, and add these to your security manager; or do not sync users and groups." +
+                                                            "Unfortunately, the way Jackrabbit goes about certain things requires us to do a bit of hacking in order to sync Authorizables securely, and efficiently.")
+        }
+        Authorizable authorizable = findAuthorizable(session)
+        if(!authorizable) {
```

Comment:
Usually I'd agree for assignments like this, but I don't believe this will work as I need to feed the result of the ternary evaluation into the else condition (updateAuthorizable)